docs/proposals/PROPOSAL_D.md

```diff
@@ -0,0 +1,519 @@
+        "relation": "contains"
+      },
+      {
+        "source": "hardware",
+        "target": "hardware.gpu",
+        "relation": "contains"
+      },
+      {
+        "source": "hardware.gpu",
+        "target": "hardware.gpu.enabled",
+        "relation": "contains"
+      },
+      {
+        "source": "hardware.gpu",
+        "target": "hardware.gpu.driver",
+        "relation": "contains"
+      },
+      {
+        "source": "hardware.gpu.driver",
+        "target": "hardware.gpu.driver.version",
+        "relation": "contains"
+      },
+      {
+        "source": "hardware.gpu",
+        "target": "hardware.gpu.architecture",
+        "relation": "contains"
+      },
+      {
+        "source": "hardware.gpu",
+        "target": "hardware.gpu.version",
+        "relation": "contains"
+      },
+      {
+        "source": "hardware.gpu",
+        "target": "hardware.gpu.cuda",
+        "relation": "contains"
+      },
+      {
+        "source": "hardware.gpu.cuda",
+        "target": "hardware.gpu.cuda.version",
+        "relation": "contains"
+      },
+      {
+        "source": "communication",
+        "target": "communication.framework",
+        "relation": "contains"
+      }
+    ],
+    "metadata": {
+      "version": "0.0.0",
+      "source": "https://github.com/supercontainers/compspec"
+    }
+  }
+}
+```
+
+## Compatibility Artifact
+
+The compatibility artifact would include a subset of metadata attributes that are known to a schema, along with versions and links to the schemas used.
+
+### Example Compatibility Artifact
+
+From the example above we might derive the following compatibility artifact.
+
+```json
+{
+  "version": "0.0.0",
+  "kind": "CompatibilitySpec",
+  "metadata": {
+    "name": "lammps-prototype",
+    "jsonSchema": "https://raw.githubusercontent.com/supercontainers/compspec/main/supercontainers/compspec.json"
+  },
+  "compatibilities": [
+    {
+      "name": "org.supercontainers",
+      "version": "0.0.0",
+      "annotations": {
+        "mpi.implementation": "mpich",
+        "mpi.version": "4.1.1",
+        "os.name": "Ubuntu 22.04.3 LTS",
+        "os.release": "22.04.3",
+        "os.vendor": "ubuntu",
+        "os.version": "22.04",
+        "hardware.gpu.available": "yes"
+      }
+    }
+  ]
+}
+```
+
+Note that we could easily add another compatibility interest group namespace, e.g, [archspec.io](https://github.com/supercontainers/compspec/blob/main/archspec/compspec.json):
+
+```json
+{
+  "version": "0.0.0",
+  "kind": "CompatibilitySpec",
+  "metadata": {
+    "name": "lammps-prototype"
+  },
+  "compatibilities": [
+    {
+      "name": "io.archspec",
+      "version": "0.0.0",
+      "schema": "https://raw.githubusercontent.com/supercontainers/compspec/main/archspec/compspec.json",
+      "annotations": {
+        "cpu.model": "13th Gen Intel(R) Core(TM) i5-1335U",
+        "cpu.target": "amd64",
+        "cpu.vendor": "GenuineIntel"
+      }
+    },
+    {
```

Comment:
if there are mutiple compatibilities list here, should meet all compatibilities, or just meet one of the compaibility, or some of them.

Reply:
That is up to the tool using the artifact. The metadata is provided with flexibility in mind.